src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/JvmGenerationTuningPolicy.java

```diff
@@ -103,35 +104,52 @@ public int increaseYoungGeneration() {
     if (currentRatio < 3) {
       return -1;
     }
-    return (int) Math.floor(currentRatio - 1);
-  }
-
-  private int getOversizedThreshold() {
-    return issueCountThreshold;
+    double newRatio = currentRatio > 5 ? 3 : currentRatio - 1;
+    return (int) Math.floor(newRatio);
   }
 
-  private int getUndersizedThreshold() {
-    return issueCountThreshold;
+  public void initializeSlidingWindows(JvmGenerationTuningPolicyConfig policyConfig) {
+    if (this.tooSmallIssues == null) {
+      this.tooSmallIssues = new PersistableSlidingWindow(policyConfig.getSlidingWindowSizeInSeconds(),
+          policyConfig.getBucketSizeInSeconds(),
+          TimeUnit.SECONDS,
+          UNDERSIZED_DATA_FILE_PATH);
+    }
+    if (this.tooLargeIssues == null) {
+      this.tooLargeIssues = new PersistableSlidingWindow(policyConfig.getSlidingWindowSizeInSeconds(),
+          policyConfig.getBucketSizeInSeconds(),
+          TimeUnit.SECONDS,
+          OVERSIZED_DATA_FILE_PATH);
+    }
   }
 
   public List<Action> actions() {
+    if (rcaConf == null || appContext ==  null) {
+      LOG.error("rca conf/app context is null, return empty action list");
+      return new ArrayList<>();
+    }
+    policyConfig = rcaConf.getDeciderConfig().getJvmGenerationTuningPolicyConfig();
+    initializeSlidingWindows(policyConfig);
     List<Action> actions = new ArrayList<>();
-    if (tooLargeIssues.size() > getOversizedThreshold()) {
+    if (tooLargeIssues.readCurrentBucket() > policyConfig.getOversizedbucketHeight()) {
       int newRatio = decreaseYoungGeneration();
       if (newRatio >= 1) {
-        actions.add(new ModifyJvmGenerationAction(appContext, decreaseYoungGeneration(), COOLOFF_PERIOD_IN_MILLIS, true));
+        actions.add(new ModifyJvmGenerationParams(appContext, decreaseYoungGeneration(), COOLOFF_PERIOD_IN_MILLIS, true));
       }
-    } else if (tooSmallIssues.size() > getUndersizedThreshold()) {
+    } else if (tooSmallIssues.readCurrentBucket() > policyConfig.getUndersizedbucketHeight()) {
```

Comment:
I'm a bit confused here. IIUC, `readCurrentBucket()` only returns the size of current hour bucket. If we only need to decide based on current hour values, then why are we saving the full sliding window?